Framework/DataHandling/src/GenerateGroupingPowder.cpp

```diff
@@ -244,7 +239,7 @@ void GenerateGroupingPowder::init() {
   fileExtensionXMLorHDF5->addAllowedValue(std::string("nxs"));
   fileExtensionXMLorHDF5->addAllowedValue(std::string("nx5"));
   declareProperty("FileFormat", std::string("xml"), fileExtensionXMLorHDF5,
-                  "File extension/format for saving output: either xml (default) or nxs/nx5.", PropertyMode::Optional);
+                  "File extension/format for saving output: either xml (default) or nxs/nx5.");
```

Comment:
I think this should remain optional.  The original work that generated this PR was to make the file optional and allow output from a workspace instead, without reading/writing from disk.

Reply:
The file is only written if `GroupingFilename` is defined and is not dependent on `FileFormat`. `FileFormat` was set to a default of `xml` which means it wasn't optional to begin with. Having set `PropertyMode::Optional` actually caused this to be incorrectly characterized as an Output property, see https://docs.mantidproject.org/nightly/algorithms/GenerateGroupingPowder-v1.html This prevented it appearing in the algorithm dialog. So that is what this change is fixing.


This still works
```python
GenerateGroupingPowder(ws,10, GroupingWorkspace="groups2")
```

Dialog before and after
![2024-07-18-110943_445x349_scrot](https://github.com/user-attachments/assets/4034cf0a-3ac5-4688-b047-649322f22d85)
![2024-07-18-111000_530x379_scrot](https://github.com/user-attachments/assets/454278a9-69aa-48bb-97a4-a2e6754695a2)

Reply:
We determined this property should be removed entirely, as it is redundant, given the `GroupingFilename` propetty.

Reply:
It's removed from version 2. We should leave it in version 1 so that has the same behaviour.